core/src/commonMain/kotlin/Peripheral.kt

```diff
@@ -0,0 +1,44 @@
+@file:Suppress("RedundantUnitReturnType")
+
+package com.juul.kable
+
+import kotlinx.coroutines.flow.Flow
+
+public expect class Peripheral {
+
+    public val state: Flow<State>
+    public val events: Flow<Event>
+
+    public val services: List<Service>?
```

Comment:
`null` represents the state where services have not yet been discovered and allows library consumer to differentiate between a lack of service discovery vs. service having been discovered but none were found on peripheral (i.e. empty list).

Reply:
Ok I see the reasoning. Do you think this would this be better represented concretely as part of the `state` rather than implied via a `null` value here?

Reply:
I think service discovery could be added as a state if we find the need; I'd like to keep that minimal (and omit that state) for now.

Perhaps adding better documentation on this property would help provide clarity around the nullness of it (I'll do that).

It's also perhaps not as common to use the notion of `null` on iOS? Unless I'm [misinterpreting the guidance](https://github.com/Kotlin/KEEP/blob/master/proposals/stdlib/result.md#error-handling-style-and-exceptions), in Kotlin, it's a bit more encouraged.

Reply:
> I think service discovery could be added as a state if we find the need; I'd like to keep that minimal (and omit that state) for now.

Here's how I see it: "that state" is not being "omitted" - it's being encapsulated as a hidden (undocumented/implied) contract behind the `null` value. Probably it's ok, most developers are going to grok that a `null` here actually has the meaning "services are not discovered yet".

I've been steeped in value types too long I suppose. Nowadays when I see `null` it's a glaringly obvious type override with an implied contract that we all went along with for 40 years. :-) I forget all this stuff from the good old days of classes and nulls!